### PR TITLE
Enhance documentation for autologging supported libraries

### PR DESCRIPTION
**Pull request by xq-yin**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/xq-yin/mlflow/pull/12356?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12356/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12356
```

</p>
</details>

### What changes are proposed in this pull request?

image
We are missing langchain, open ai, pyspark.ml, paddle and the mention of tensorflow (tho the usage is described in keras section) in the list. Adding these to the doc.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

`make clean livehtml`
Screenshot 2024-06-14 at 3 33 00 PM
Screenshot 2024-06-14 at 3 33 06 PM
Screenshot 2024-06-14 at 3 33 30 PM
Screenshot 2024-06-14 at 3 33 35 PM

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/tracking/autolog.rst [hunk 1]
```diff
@@ -176,8 +176,8 @@ Autologging captures the following information:
 
 .. _autolog-keras:
 
-Keras
------
+Keras/TensorFlow
+----------------
 Call the generic autolog function or :py:func:`mlflow.tensorflow.autolog` before your training code to enable automatic logging of metrics and parameters. As an example, try running the `Keras/Tensorflow example <https://github.com/mlflow/mlflow/blob/master/examples/keras/train.py>`_.
```
Comment (xq-yin) [inline docs/source/tracking/autolog.rst:181]: did not create a separate section for TensorFlow cuz it's going to be a pure copy-paste of this section. decided to merge them together. let me know what everyone thinks

File: docs/source/tracking/autolog.rst [hunk 2]
```diff
@@ -201,6 +201,25 @@ Also, MLflow will then automatically end the run once training ends via calls to
...
+
+Autologging captures the following information:
+
++-----------+-------------+----------------+---------------+---------------------------------------------------------------------------------+
+| Framework | Metrics     | Parameters     | Tags          | Artifacts                                                                       |
++-----------+-------------+----------------+---------------+---------------------------------------------------------------------------------+
+| LangChain | --          | --             |  --           | - Traces                                                                        |
+|           |             |                |               | - `MLflow Model`_ (LangChain model) with model signature on training end        |
+|           |             |                |               | - Input example                                                                 |
+|           |             |                |               | - Datasets                                                                      |
```
Comment (serena-ruan) [inline docs/source/tracking/autolog.rst:220]: Honestly langchain autologging doesn't log datasets 😆 We should remove it from code as well

Comment (xq-yin) [reply docs/source/tracking/autolog.rst:220]: oh! FYI @B-Step62 for the langchain doc page as well

Comment (B-Step62) [reply docs/source/tracking/autolog.rst:220]: Ooh lol I didn't even realize I mentioned datasets there, it's almost auto-complete for me typing `models, datasets, etc...`. Will open a quick PR.

File: docs/source/tracking/autolog.rst [hunk 3]
```diff
@@ -219,6 +238,56 @@ If early stopping is activated, metrics at the best iteration will be logged as
...
+See an example usage with `OpenAI <https://github.com/mlflow/mlflow/tree/master/examples/openai>`_.
+
+Autologging captures the following information:
+
++-----------+-------------+----------------+---------------+------------------------------------------------------------------------------+
+| Framework | Metrics     | Parameters     | Tags          | Artifacts                                                                    |
++-----------+-------------+----------------+---------------+------------------------------------------------------------------------------+
+| OpenAI    | --          | --             |  --           | - `MLflow Model`_ (OpenAI model) with model signature on training end        |
+|           |             |                |               | - Input example                                                              |
+|           |             |                |               | - Datasets                                                                   |
```
Comment (xq-yin) [inline docs/source/tracking/autolog.rst:256]: @serena-ruan do you know if datasets are logged here? I copy-paste from langchain without a second thought, cuz the code looks quite similar

Comment (serena-ruan) [reply docs/source/tracking/autolog.rst:256]: Seems no, and dataset actually doesn't make so much sense for these two flavors, I'll remove them

Comment (B-Step62) [reply docs/source/tracking/autolog.rst:256]: Actually there are a few more flavors that has `log_datasets` in `autolog()` but no-op e.g. `gluon`, `fastai`, `pytorch`, etc. Probably just a copy-and-paste thing, but might be safer to ask team about any historical context.

Comment (xq-yin) [reply docs/source/tracking/autolog.rst:256]: I see. actually datasets are not mentioned in the doc for other flavours (phew). we could make that change in a separate PR

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Review by B-Step62 — APPROVED** [review]

File: docs/source/tracking/autolog.rst [hunk 4]
```diff
@@ -201,6 +201,24 @@ Also, MLflow will then automatically end the run once training ends via calls to
 
 If a run already exists when ``autolog()`` captures data, MLflow will log to that run but not automatically end that run after training. You will have to manually stop the run if you wish to start a new run context for logging to a new run.
 
+.. _autolog-langchain:
+
+LangChain
+---------
+
+Call the generic autolog function :py:func:`mlflow.langchain.autolog` before your training code to enable automatic logging of traces.
+See an example usage with `LangChain <https://github.com/mlflow/mlflow/tree/master/examples/langchain>`_.
```
[the remark this answers is not in the document]
Comment (serena-ruan) [reply docs/source/tracking/autolog.rst:210]: @xq-yin Let's address this comment in a follow-up PR :)

[the remark this answers is not in the document]
Comment (xq-yin) [reply docs/source/tracking/autolog.rst:210]: @B-Step62 the :py:func:`mlflow.langchain.autolog` actually links to the lc autologging doc page. hence I followed other sections to add examples here. do you think it's better to explicitly link to autologging doc page again? if needed I can just remove the example usage part as well

[the remark this answers is not in the document]
Comment (B-Step62) [reply docs/source/tracking/autolog.rst:210]: @xq-yin I mean the `docs/source/llms/langchain/autologging.rst` page not the API doc page. Other flavors don't have such standalone page for autologging.

**Pull request merged**